rust/src/offer.rs

```diff
@@ -13,29 +10,32 @@ pub struct Offer {
     pub index: f64,
 }
 
-pub async fn get_offer() -> Result<Option<Offer>> {
+pub async fn get_offer() -> Option<Offer> {
     let client = reqwest::Client::builder()
         .timeout(crate::config::TCP_TIMEOUT)
-        .build()?;
+        .build()
+        .expect("reqwest client to build");
     let result = client.get(maker_endpoint() + "/api/offer").send().await;
     let response = match result {
         Ok(res) => res,
         Err(err) => {
             tracing::error!("Could not fetch offers {err:?}");
-            return Ok(None);
+            return None;
         }
     };
 
     if response.status() == StatusCode::NOT_FOUND
         || response.status() == StatusCode::INTERNAL_SERVER_ERROR
     {
-        let response = response.text().await?;
-        bail!("Failed to fetch offer: {response}")
+        tracing::warn!("Failed to fetch offer");
```

Comment:
🔧 The response context might still be useful for the log.

Reply:
https://github.com/itchysats/10101/pull/550/commits/a0ea0da977c31ccd59c1a12da7f58b684100b1bc